src/rhash/hash.c

```diff
@@ -679,6 +679,325 @@ static int rc_hash_7800(char hash[33], const uint8_t* buffer, size_t buffer_size
+  const uint8_t  RC_ZIP_CDH_FILENAME_LEN_OFS = 28;
+  const uint8_t  RC_ZIP_CDH_EXTRA_LEN_OFS = 30;
+  const uint8_t  RC_ZIP_CDH_LOCAL_HEADER_OFS = 42;
+  const uint8_t  RC_ZIP_CDH_COMMENT_LEN_OFS = 32;
+  const uint8_t  RC_ZIP_LOCAL_DIR_HEADER_SIZE = 30;
+  const uint32_t RC_ZIP64_END_OF_CENTRAL_DIR_LOCATOR_SIG = 0x07064b50;
+  const uint8_t  RC_ZIP64_END_OF_CENTRAL_DIR_HEADER_SIZE = 56;
+  const uint8_t  RC_ZIP64_END_OF_CENTRAL_DIR_LOCATOR_SIZE = 20;
+  const uint8_t  RC_ZIP64_ECDL_REL_OFS_TO_ZIP64_ECDR_OFS = 8;
+  const uint32_t RC_ZIP64_END_OF_CENTRAL_DIR_HEADER_SIG = 0x06064b50;
+  const uint8_t  RC_ZIP64_ECDH_CDIR_TOTAL_ENTRIES_OFS = 32;
+  const uint8_t  RC_ZIP64_ECDH_CDIR_SIZE_OFS = 40;
+  const uint8_t  RC_ZIP64_ECDH_CDIR_OFS_OFS = 48;
+  const size_t sizeof_idx = sizeof(struct rc_hash_zip_idx);
+
+  /* Macros used for reading ZIP and writing to a buffer for hashing (undefined again at the end of the function) */
+  #define RC_ZIP_READ_LE16(p) ((uint16_t)(((const uint8_t*)(p))[0]) | ((uint16_t)(((const uint8_t*)(p))[1]) << 8U))
+  #define RC_ZIP_READ_LE32(p) ((uint32_t)(((const uint8_t*)(p))[0]) | ((uint32_t)(((const uint8_t*)(p))[1]) << 8U) | ((uint32_t)(((const uint8_t*)(p))[2]) << 16U) | ((uint32_t)(((const uint8_t*)(p))[3]) << 24U))
+  #define RC_ZIP_READ_LE64(p) ((uint64_t)(((const uint8_t*)(p))[0]) | ((uint64_t)(((const uint8_t*)(p))[1]) << 8U) | ((uint64_t)(((const uint8_t*)(p))[2]) << 16U) | ((uint64_t)(((const uint8_t*)(p))[3]) << 24U) | ((uint64_t)(((const uint8_t*)(p))[4]) << 32U) | ((uint64_t)(((const uint8_t*)(p))[5]) << 40U) | ((uint64_t)(((const uint8_t*)(p))[6]) << 48U) | ((uint64_t)(((const uint8_t*)(p))[7]) << 56U))
+  #define RC_ZIP_WRITE_LE32(p,v) { ((uint8_t*)(p))[0] = (uint8_t)((uint32_t)(v) & 0xFF); ((uint8_t*)(p))[1] = (uint8_t)(((uint32_t)(v) >> 8) & 0xFF); ((uint8_t*)(p))[2] = (uint8_t)(((uint32_t)(v) >> 16) & 0xFF); ((uint8_t*)(p))[3] = (uint8_t)((uint32_t)(v) >> 24); }
+  #define RC_ZIP_WRITE_LE64(p,v) { ((uint8_t*)(p))[0] = (uint8_t)((uint64_t)(v) & 0xFF); ((uint8_t*)(p))[1] = (uint8_t)(((uint64_t)(v) >> 8) & 0xFF); ((uint8_t*)(p))[2] = (uint8_t)(((uint64_t)(v) >> 16) & 0xFF); ((uint8_t*)(p))[3] = (uint8_t)(((uint64_t)(v) >> 24) & 0xFF); ((uint8_t*)(p))[4] = (uint8_t)(((uint64_t)(v) >> 32) & 0xFF); ((uint8_t*)(p))[5] = (uint8_t)(((uint64_t)(v) >> 40) & 0xFF); ((uint8_t*)(p))[6] = (uint8_t)(((uint64_t)(v) >> 48) & 0xFF); ((uint8_t*)(p))[7] = (uint8_t)((uint64_t)(v) >> 56); }
+
+  uint8_t buf[2048], *alloc_buf, *cdir_start, *cdir_max, *cdir, *hashdata;
+  uint32_t cdir_entry_len;
+  size_t indices_offset, alloc_size;
+  int64_t i_file, archive_size, ecdh_ofs, total_files, cdir_size, cdir_ofs;
+  struct rc_hash_zip_idx* hashindices, *hashindex;
+
+  rc_file_seek(file_handle, 0, SEEK_END);
+  archive_size = rc_file_tell(file_handle);
+
+  /* Basic sanity checks - reject files which are too small. */
+  if (archive_size < RC_ZIP_END_OF_CENTRAL_DIR_HEADER_SIZE)
+  {
+    rc_file_close(file_handle);
```

Comment:
It looks like the calling function will close the file handle. (applies several times)

Reply:
Fixed in the latest commit. Yes the caller is in charge of opening and closing the file handle.